packages/x-scheduler/src/material/timeline/content/TimelineContent.tsx

```diff
@@ -0,0 +1,155 @@
+'use client';
+import * as React from 'react';
+import clsx from 'clsx';
+import { useStore } from '@base-ui-components/utils/store';
+import { getColorClassName } from '../../internals/utils/color-utils';
+import { DEFAULT_EVENT_COLOR } from '../../../primitives/utils/SchedulerStore';
+import { Timeline as TimelinePrimitive } from '../../../primitives/timeline';
+import { TimelineEvent } from './timeline-event';
+import { selectors } from '../../../primitives/use-timeline';
+import { useTimelineStoreContext } from '../../../primitives/utils/useTimelineStoreContext';
+import { useEventOccurrencesWithTimelinePositionForMap } from '../../../primitives/use-event-occurrences-with-timeline-position';
+import {
+  DaysHeader,
+  MonthsHeader,
+  TimeHeader,
+  WeeksHeader,
+  YearHeader,
+} from './header-toolbar/view-header';
+import { useEventOccurrencesGroupedByResource } from '../../../primitives/use-event-occurrences-grouped-by-resource';
+import { getAdapter } from '../../../primitives/utils/adapter/getAdapter';
+import { diffIn } from '../../..//primitives/utils/date-utils';
+import { SchedulerValidDate } from '../../../primitives';
+import { TimelineContentProps } from './TimelineContent.types';
+import { TimelineView } from '../Timeline.types';
+
+const adapter = getAdapter();
+
+const getEndBoundaries = (view: TimelineView, start: SchedulerValidDate) => {
+  const endBoundaries = {
+    time: adapter.addHours(start, 72),
+    days: adapter.addDays(start, 21),
+    weeks: adapter.addWeeks(start, 8),
+    months: adapter.addMonths(start, 12),
+    years: adapter.addYears(start, 4),
+  };
+
+  return endBoundaries[view];
+};
+
+type UnitType = 'hours' | 'days' | 'weeks' | 'months' | 'years';
+
+const UNIT: { [key: string]: UnitType } = {
+  time: 'hours',
+  days: 'days',
+  weeks: 'weeks',
+  months: 'months',
+  years: 'years',
+};
+
+export const TimelineContent = React.forwardRef(function TimelineContent(
+  props: TimelineContentProps,
+  forwardedRef: React.ForwardedRef<HTMLDivElement>,
+) {
+  const { view = 'days', className, ...other } = props;
+  const store = useTimelineStoreContext();
+  const resources = useStore(store, selectors.resources);
+  const visibleDate = useStore(store, selectors.visibleDate);
+
+  const start = visibleDate;
+  const end = React.useMemo(() => getEndBoundaries(view, start), [view, start]);
+
+  const occurrencesMap = useEventOccurrencesGroupedByResource({
+    start,
+    end,
+  });
+  const resourcesWithOccurrences = useEventOccurrencesWithTimelinePositionForMap({
+    occurrences: occurrencesMap,
+    maxColumnSpan: Infinity,
```

Comment:
Do we want column spanning?

<img width="1821" height="299" alt="Image" src="https://github.com/user-attachments/assets/22b0e54f-7850-4a90-8f15-b4957d1f73a2" />

Here, the event 6 doesn't span

Btw I should rename it "maxSpan" because on the timeline it's a row spanning